app/components/UI/Perps/Views/PerpsClosePositionView/PerpsClosePositionView.tsx

```diff
@@ -0,0 +1,653 @@
+  const isLong = parseFloat(position.size) > 0;
+  const absSize = Math.abs(parseFloat(position.size));
+
+  // Calculate position value and effective margin
+  const positionValue = absSize * currentPrice;
+  // Use the actual margin from the position instead of recalculating
+  const effectiveMargin = parseFloat(position.marginUsed);
+
+  // Use unrealized PnL from position
+  const pnl = parseFloat(position.unrealizedPnl);
+
+  // Calculate fees using the unified fee hook
+  const closingValue = positionValue * (closePercentage / 100);
+  const feeResults = usePerpsOrderFees({
+    orderType,
+    amount: closingValue.toString(),
+    isMaker: false, // Closing positions are typically taker orders
+  });
+
+  // Calculate what user will receive (effective margin + pnl - fees)
+  const receiveAmount =
+    (closePercentage / 100) * effectiveMargin +
+    (closePercentage / 100) * pnl -
+    feeResults.totalFee;
+
+  // Get minimum order amount for this asset
+  const { minimumOrderAmount } = useMinimumOrderAmount({
+    asset: position.coin,
+  });
+
+  // Calculate remaining position value after partial close
+  const remainingPositionValue = positionValue * (1 - closePercentage / 100);
+  const isPartialClose = closePercentage < 100;
+
+  // Use the validation hook
+  const validationResult = usePerpsClosePositionValidation({
+    coin: position.coin,
+    closePercentage,
+    closeAmount: closeAmount.toString(),
+    orderType,
+    limitPrice,
+    currentPrice,
+    positionSize: absSize,
+    positionValue,
+    minimumOrderAmount,
+    closingValue,
+    remainingPositionValue,
+    receiveAmount,
+    isPartialClose,
+  });
+
+  const { handleClosePosition, isClosing } = usePerpsClosePosition({
+    onSuccess: () => {
+      // Positions update automatically via WebSocket
+      navigation.goBack();
+    },
+  });
+
+  // Track position close screen viewed event
+  useEffect(() => {
+    if (!hasTrackedCloseView.current) {
+      track(MetaMetricsEvents.PERPS_POSITION_CLOSE_SCREEN_VIEWED, {
+        [PerpsEventProperties.ASSET]: position.coin,
+        [PerpsEventProperties.DIRECTION]: isLong
+          ? PerpsEventValues.DIRECTION.LONG
+          : PerpsEventValues.DIRECTION.SHORT,
+        [PerpsEventProperties.POSITION_SIZE]: absSize,
+        [PerpsEventProperties.UNREALIZED_PNL_DOLLAR]: pnl,
+      });
+      hasTrackedCloseView.current = true;
+    }
+  }, [position.coin, isLong, absSize, pnl, track]);
+
+  // Auto-open limit price bottom sheet when switching to limit order
+  useEffect(() => {
+    if (orderType === 'limit' && !limitPrice) {
+      setIsLimitPriceVisible(true);
+    }
+  }, [orderType, limitPrice]);
+
+  // Update close amount when percentage changes
+  useEffect(() => {
+    const newAmount = (closePercentage / 100) * absSize;
+    setCloseAmount(newAmount.toString());
+    setCloseAmountUSD(newAmount * currentPrice);
+
+    // Track position close value changed
+    if (closePercentage !== 100) {
```

Comment:
Using constant for `100` here would read better 
```suggestion
    if (closePercentage !== DEFAULT_CLOSE_PERCENTAGE) {
```